test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -177,6 +182,66 @@ public void initClient() throws IOException {
         assert hasXPack != null;
         assert nodeVersions != null;
     }
+    
+    // Helper class to check warnings in REST responses with sensitivity to versions
+    // used in the target cluster.
+    protected class VersionSensitiveWarningsHandler implements WarningsHandler {
+        Set<String> requiredSameVersionClusterWarnings = new HashSet<>();
+        Set<String> allowedWarnings = new HashSet<>();
+
+        public void current(String... requiredWarnings) {
+            requiredSameVersionClusterWarnings.addAll(Arrays.asList(requiredWarnings));
+            if(clusterHasACurrentVersionNode()) {
```

Comment:
Do we need this if statement actually?

Reply:
I did wonder about that too. Let's drop it.